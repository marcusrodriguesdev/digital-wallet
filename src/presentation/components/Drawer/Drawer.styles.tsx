import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${({ theme }) => theme?.colors?.background ?? '#242629'};
  border-right: 1px solid ${({ theme }) => theme?.colors?.cardBackground ?? '#16161a'};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
  padding: 40px 0 40px 32px;
  width: 270px;
`

export const Wrapper = styled.div``

export const TitleContainer = styled.div`
  display: flex;
  justify-content: space-between;
`

export const Title = styled.h1`
  ${({ theme }) => theme?.fonts?.title ?? {}}
  color: ${({ theme }) => theme?.colors?.highlight ?? '#7F5AF0'};
`

export const CloseButton = styled.button`
  align-items: center;
  background-color: ${({ theme }) => theme?.colors?.background ?? '#242629'};
  border-bottom: 0.5px solid ${({ theme }) => theme?.colors?.cardBackground ?? '#16161a'};
  border-bottom-left-radius: 8px;
  border-left: 0.5px solid ${({ theme }) => theme?.colors?.cardBackground ?? '#16161a'};
  border-top: 0.5px solid ${({ theme }) => theme?.colors?.cardBackground ?? '#16161a'};
  border-top-left-radius: 8px;
  border-right: none;
  cursor: pointer;
  display: flex;
  height: 28px;
  justify-content: center;
  outline: none;
  width: 34px;
`

export const OverviewContainer = styled.div`
  display: flex;
  flex-direction: column;
`

export const Text = styled.p`
  ${({ theme }) => theme?.fonts?.infoSelected ?? {}}
  color: ${({ theme }) => theme?.colors?.subHeadline ?? '#94a1b2'};
  margin-bottom: 16px;
  margin-top: 50px;
`

interface ButtonProps {
  selected: boolean
}
export const Button = styled.button<ButtonProps>`
  ${({ theme }) => theme?.fonts?.info ?? {}}
  align-items: center;
  color: ${({ theme, selected }) =>
    selected ? theme?.colors?.background ?? '#242629' : theme?.colors?.subHeadline ?? '#94a1b2'};
  cursor: pointer;
  background-color: ${({ theme, selected }) =>
    selected ? theme?.colors?.highlight ?? '#7f5af0' : theme?.colors?.background ?? '#242629'};
  border: none;
  border-radius: 8px;
  display: flex;
  height: 56px;
  margin-bottom: 4px;
  margin-left: -16px;
  padding-left: 16px;
  width: 237px;

  img {
    background-color: transparent;
    margin-right: 16px;
  }
`
