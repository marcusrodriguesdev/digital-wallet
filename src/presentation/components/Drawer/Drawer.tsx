import React from 'react'

import { Assets } from '@/presentation/theme/Assets'

import {
  CloseButton,
  Container,
  OverviewContainer,
  Title,
  TitleContainer,
  Text,
  Button,
  Wrapper
} from './Drawer.styles'

export interface DrawerProps {
  dashboardSelected: boolean
  settingSelected: boolean
  walletSelected: boolean
  logoutSelected: boolean
  onCloseMenu: () => void
  onSelectWallet: () => void
  onSelectSetting: () => void
  onLogoutSelected: () => void
  onSelectDashboard: () => void
}

const Drawer: React.FC<DrawerProps> = (props: DrawerProps) => {
  const {
    dashboardSelected,
    settingSelected,
    walletSelected,
    logoutSelected,
    onCloseMenu,
    onSelectWallet,
    onSelectSetting,
    onLogoutSelected,
    onSelectDashboard
  } = props

  return (
    <Container>
      <Wrapper>
        <TitleContainer>
          <Title>Digital Wallet</Title>
          <CloseButton onClick={onCloseMenu} type="button" data-testid="Drawer/CloseButton">
            <img src={Assets.ArrowLeft} alt="voltar" />
          </CloseButton>
        </TitleContainer>
        <OverviewContainer>
          <Text>VISÃO GERAL</Text>
          <Button selected={walletSelected} onClick={onSelectWallet} type="button" data-testid="Drawer/WalletButton">
            <img src={walletSelected ? Assets.WalletSelected : Assets.Wallet} alt="voltar" />
            Carteira
          </Button>
          <Button
            selected={dashboardSelected}
            onClick={onSelectDashboard}
            type="button"
            data-testid="Drawer/DashboardButton">
            <img src={dashboardSelected ? Assets.DashboardSelected : Assets.Dashboard} alt="painel de controle" />
            Painel de Controle
          </Button>
        </OverviewContainer>
      </Wrapper>
      <Wrapper>
        <OverviewContainer>
          <Text>OUTROS</Text>
          <Button
            selected={settingSelected}
            onClick={onSelectSetting}
            type="button"
            data-testid="Drawer/SettingsButton">
            <img src={settingSelected ? Assets.SettingsSelected : Assets.Settings} alt="configurações" />
            Configurações
          </Button>
          <Button selected={logoutSelected} onClick={onLogoutSelected} type="button" data-testid="Drawer/LogoutButton">
            <img src={logoutSelected ? Assets.LogoutSelected : Assets.Logout} alt="sair" />
            Sair
          </Button>
        </OverviewContainer>
      </Wrapper>
    </Container>
  )
}

export { Drawer }
