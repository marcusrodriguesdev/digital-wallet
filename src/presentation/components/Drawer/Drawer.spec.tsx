import React from 'react'

import { render } from '@testing-library/react'

import { Drawer, DrawerProps } from './Drawer'

const makeSut = (props?: Partial<DrawerProps>): any => {
  const dashboardSelected = props?.dashboardSelected ?? true
  const settingSelected = props?.settingSelected ?? false
  const walletSelected = props?.walletSelected ?? false
  const logoutSelected = props?.logoutSelected ?? false
  const onCloseMenu = props?.onCloseMenu ?? jest.fn()
  const onSelectWallet = props?.onSelectWallet ?? jest.fn()
  const onSelectSetting = props?.onSelectSetting ?? jest.fn()
  const onLogoutSelected = props?.onLogoutSelected ?? jest.fn()
  const onSelectDashboard = props?.onSelectDashboard ?? jest.fn()

  const sut = (
    <Drawer
      dashboardSelected={dashboardSelected}
      settingSelected={settingSelected}
      walletSelected={walletSelected}
      logoutSelected={logoutSelected}
      onCloseMenu={onCloseMenu}
      onSelectWallet={onSelectWallet}
      onSelectSetting={onSelectSetting}
      onLogoutSelected={onLogoutSelected}
      onSelectDashboard={onSelectDashboard}
    />
  )

  return {
    sut,
    dashboardSelected,
    settingSelected,
    walletSelected,
    logoutSelected,
    onCloseMenu,
    onSelectWallet,
    onSelectSetting,
    onLogoutSelected,
    onSelectDashboard
  }
}

describe('presentation/components/Drawer', () => {
  it('should match snapshot', () => {
    const { sut } = makeSut()
    const component = render(sut)
    expect(component).toMatchSnapshot()
  })

  it('should render other icons accordingly', () => {
    const { sut } = makeSut({
      dashboardSelected: false,
      settingSelected: true,
      walletSelected: true,
      logoutSelected: true
    })
    const component = render(sut)
    expect(component).toMatchSnapshot()
  })

  it('should call onSelectDashboard when dashboard button is clicked', () => {
    const { sut, onSelectDashboard } = makeSut()

    const { getByTestId } = render(sut)
    const button = getByTestId('Drawer/DashboardButton')

    button.click()

    expect(onSelectDashboard).toBeCalledTimes(1)
  })

  it('should call onSelectWallet when wallet button is clicked', () => {
    const { sut, onSelectWallet } = makeSut()

    const { getByTestId } = render(sut)
    const button = getByTestId('Drawer/WalletButton')

    button.click()

    expect(onSelectWallet).toBeCalledTimes(1)
  })

  it('should call onSelectSetting when setting button is clicked', () => {
    const { sut, onSelectSetting } = makeSut()

    const { getByTestId } = render(sut)
    const button = getByTestId('Drawer/SettingsButton')

    button.click()

    expect(onSelectSetting).toBeCalledTimes(1)
  })

  it('should call onLogoutSelected when logout button is clicked', () => {
    const { sut, onLogoutSelected } = makeSut()

    const { getByTestId } = render(sut)
    const button = getByTestId('Drawer/LogoutButton')

    button.click()

    expect(onLogoutSelected).toBeCalledTimes(1)
  })

  it('should call onCloseMenu when close button is clicked', () => {
    const { sut, onCloseMenu } = makeSut()

    const { getByTestId } = render(sut)
    const button = getByTestId('Drawer/CloseButton')

    button.click()

    expect(onCloseMenu).toBeCalledTimes(1)
  })
})
