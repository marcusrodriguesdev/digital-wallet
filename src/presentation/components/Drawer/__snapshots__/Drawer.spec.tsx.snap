// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentation/components/Drawer should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hHHAib"
      >
        <div
          class="sc-gswNZR"
        >
          <div
            class="sc-dkrFOg cLdviv"
          >
            <h1
              class="sc-hLBbgP iFiqwX"
            >
              Digital Wallet
            </h1>
            <button
              class="sc-eDvSVe jObgSB"
              data-testid="Drawer/CloseButton"
              type="button"
            >
              <img
                alt="voltar"
                src="[object Object]"
              />
            </button>
          </div>
          <div
            class="sc-jSUZER eeFZNB"
          >
            <p
              class="sc-gKPRtg dHBAKV"
            >
              VISÃO GERAL
            </p>
            <button
              class="sc-iBYQkv kzDncL"
              data-testid="Drawer/WalletButton"
              type="button"
            >
              <img
                alt="voltar"
                src="[object Object]"
              />
              Carteira
            </button>
            <button
              class="sc-iBYQkv jXEoQB"
              data-testid="Drawer/DashboardButton"
              type="button"
            >
              <img
                alt="painel de controle"
                src="[object Object]"
              />
              Painel de Controle
            </button>
          </div>
        </div>
        <div
          class="sc-gswNZR"
        >
          <div
            class="sc-jSUZER eeFZNB"
          >
            <p
              class="sc-gKPRtg dHBAKV"
            >
              OUTROS
            </p>
            <button
              class="sc-iBYQkv kzDncL"
              data-testid="Drawer/SettingsButton"
              type="button"
            >
              <img
                alt="configurações"
                src="[object Object]"
              />
              Configurações
            </button>
            <button
              class="sc-iBYQkv kzDncL"
              data-testid="Drawer/LogoutButton"
              type="button"
            >
              <img
                alt="sair"
                src="[object Object]"
              />
              Sair
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hHHAib"
    >
      <div
        class="sc-gswNZR"
      >
        <div
          class="sc-dkrFOg cLdviv"
        >
          <h1
            class="sc-hLBbgP iFiqwX"
          >
            Digital Wallet
          </h1>
          <button
            class="sc-eDvSVe jObgSB"
            data-testid="Drawer/CloseButton"
            type="button"
          >
            <img
              alt="voltar"
              src="[object Object]"
            />
          </button>
        </div>
        <div
          class="sc-jSUZER eeFZNB"
        >
          <p
            class="sc-gKPRtg dHBAKV"
          >
            VISÃO GERAL
          </p>
          <button
            class="sc-iBYQkv kzDncL"
            data-testid="Drawer/WalletButton"
            type="button"
          >
            <img
              alt="voltar"
              src="[object Object]"
            />
            Carteira
          </button>
          <button
            class="sc-iBYQkv jXEoQB"
            data-testid="Drawer/DashboardButton"
            type="button"
          >
            <img
              alt="painel de controle"
              src="[object Object]"
            />
            Painel de Controle
          </button>
        </div>
      </div>
      <div
        class="sc-gswNZR"
      >
        <div
          class="sc-jSUZER eeFZNB"
        >
          <p
            class="sc-gKPRtg dHBAKV"
          >
            OUTROS
          </p>
          <button
            class="sc-iBYQkv kzDncL"
            data-testid="Drawer/SettingsButton"
            type="button"
          >
            <img
              alt="configurações"
              src="[object Object]"
            />
            Configurações
          </button>
          <button
            class="sc-iBYQkv kzDncL"
            data-testid="Drawer/LogoutButton"
            type="button"
          >
            <img
              alt="sair"
              src="[object Object]"
            />
            Sair
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`presentation/components/Drawer should render other icons accordingly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hHHAib"
      >
        <div
          class="sc-gswNZR"
        >
          <div
            class="sc-dkrFOg cLdviv"
          >
            <h1
              class="sc-hLBbgP iFiqwX"
            >
              Digital Wallet
            </h1>
            <button
              class="sc-eDvSVe jObgSB"
              data-testid="Drawer/CloseButton"
              type="button"
            >
              <img
                alt="voltar"
                src="[object Object]"
              />
            </button>
          </div>
          <div
            class="sc-jSUZER eeFZNB"
          >
            <p
              class="sc-gKPRtg dHBAKV"
            >
              VISÃO GERAL
            </p>
            <button
              class="sc-iBYQkv jXEoQB"
              data-testid="Drawer/WalletButton"
              type="button"
            >
              <img
                alt="voltar"
                src="[object Object]"
              />
              Carteira
            </button>
            <button
              class="sc-iBYQkv kzDncL"
              data-testid="Drawer/DashboardButton"
              type="button"
            >
              <img
                alt="painel de controle"
                src="[object Object]"
              />
              Painel de Controle
            </button>
          </div>
        </div>
        <div
          class="sc-gswNZR"
        >
          <div
            class="sc-jSUZER eeFZNB"
          >
            <p
              class="sc-gKPRtg dHBAKV"
            >
              OUTROS
            </p>
            <button
              class="sc-iBYQkv jXEoQB"
              data-testid="Drawer/SettingsButton"
              type="button"
            >
              <img
                alt="configurações"
                src="[object Object]"
              />
              Configurações
            </button>
            <button
              class="sc-iBYQkv jXEoQB"
              data-testid="Drawer/LogoutButton"
              type="button"
            >
              <img
                alt="sair"
                src="[object Object]"
              />
              Sair
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hHHAib"
    >
      <div
        class="sc-gswNZR"
      >
        <div
          class="sc-dkrFOg cLdviv"
        >
          <h1
            class="sc-hLBbgP iFiqwX"
          >
            Digital Wallet
          </h1>
          <button
            class="sc-eDvSVe jObgSB"
            data-testid="Drawer/CloseButton"
            type="button"
          >
            <img
              alt="voltar"
              src="[object Object]"
            />
          </button>
        </div>
        <div
          class="sc-jSUZER eeFZNB"
        >
          <p
            class="sc-gKPRtg dHBAKV"
          >
            VISÃO GERAL
          </p>
          <button
            class="sc-iBYQkv jXEoQB"
            data-testid="Drawer/WalletButton"
            type="button"
          >
            <img
              alt="voltar"
              src="[object Object]"
            />
            Carteira
          </button>
          <button
            class="sc-iBYQkv kzDncL"
            data-testid="Drawer/DashboardButton"
            type="button"
          >
            <img
              alt="painel de controle"
              src="[object Object]"
            />
            Painel de Controle
          </button>
        </div>
      </div>
      <div
        class="sc-gswNZR"
      >
        <div
          class="sc-jSUZER eeFZNB"
        >
          <p
            class="sc-gKPRtg dHBAKV"
          >
            OUTROS
          </p>
          <button
            class="sc-iBYQkv jXEoQB"
            data-testid="Drawer/SettingsButton"
            type="button"
          >
            <img
              alt="configurações"
              src="[object Object]"
            />
            Configurações
          </button>
          <button
            class="sc-iBYQkv jXEoQB"
            data-testid="Drawer/LogoutButton"
            type="button"
          >
            <img
              alt="sair"
              src="[object Object]"
            />
            Sair
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
